### PR TITLE
Issue 200 - Adding Caltech Birds 2010 Dataset

### PR DESCRIPTION
**Pull request by iswariyam**
Added files for incorporating Caltech Birds 2010 dataset [Issue 200](https://github.com/tensorflow/datasets/issues/200#issuecomment-473447558)

https://gist.github.com/iswariyam/05fd230d5997e075f8f9f5d715aecab1

### CODE REVIEW

File: tensorflow_datasets/image/__init__.py [hunk 1]
```diff
@@ -41,6 +41,7 @@
 from tensorflow_datasets.image.open_images import OpenImagesV4
 from tensorflow_datasets.image.quickdraw import QuickdrawBitmap
 from tensorflow_datasets.image.rock_paper_scissors import RockPaperScissors
+from tensorflow_datasets.image.caltech_birds import CaltechBirds2010
```
Comment (us) [inline tensorflow_datasets/image/__init__.py]: you should add with name order.

File: tensorflow_datasets/image/caltech_birds.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    VERSION = tfds.core.Version("0.1.0")
+
+    def _info(self):
+
+        return tfds.core.DatasetInfo(
+                builder=self,
+                description=_DESCRIPTION,
+                features=tfds.features.FeaturesDict({
+                    # Images are of varying size
+                    # TODO: Need to add part attributes and their annotations
```
Comment (us) [inline tensorflow_datasets/image/caltech_birds.py]: You should check `tensorflow_datasets/image/coco.py` about todo.

Comment (iswariyam) [reply tensorflow_datasets/image/caltech_birds.py]: Could please elaborate on it? I looked in coco.py and did not understand it

**Comment by iswariyam** [comment]
Added 2011 dataset [issue 201](https://github.com/tensorflow/datasets/issues/201#issue-419145693) also

**Comment by iswariyam** [comment]
> **The test data should not contain any copyrighted material.**

@us  Changed the test data

File: tensorflow_datasets/image/caltech_birds_test.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    SPLITS = {  # No. of train and test samples
+      'train': 9,
+      'test': 6,
+    }
+
+    DL_EXTRACT_RESULT = ['Lists.tgz', 'Annotations.tgz', 'Images.tar.gz']
+
+
+class CaltechBirdsTest(tfds_test.DatasetBuilderTestCase):
```
Comment (us) [inline tensorflow_datasets/image/caltech_birds_test.py]: I think you forgot that? You have 2 test class. Delete it the above is better.

Comment (iswariyam) [reply tensorflow_datasets/image/caltech_birds_test.py]: Yes, I have 2 test classes for two different datasets (CaltechBirds2010 and CaltechBirds2011) similar to what has been done in cifar10 and cifar100.
Could please clarify on what fake examples I have to delete.

Comment (us) [reply tensorflow_datasets/image/caltech_birds_test.py]: Sorry I miss it 2011 but if datasets file structure are same. You can inherit like `mnist_test`

Comment (iswariyam) [reply tensorflow_datasets/image/caltech_birds_test.py]: Unfortunately the datasets do not have the same structure.

**Comment by us** [comment]
You should squash and clean commits. These commits too much for a dataset.

**Pull request merged**